src/services/UserService.ts

```diff
@@ -145,32 +147,43 @@ class UserService {
    * @param data - Data to update the user with
    */
   public async updateUser(authUser: AuthUser, data: UserInput): Promise<void> {
-    FirebaseService.getAuth().currentUser?.getIdToken();
-    return FirebaseService.doEmailUpdate(data.password, data.email).then(() => {
+    try {
+      FirebaseService.getAuth().currentUser?.getIdToken();
+      if (authUser.firebaseUser.email !== data.email) {
+        await FirebaseService.doEmailUpdate(data.password, data.email);
+      }
```

Comment:
Not updating email if its the same

Reply:
This works, but the reason I didn't check if the email is the same is because I wanted to call the doReauthenticateUser() function (called by doEmailUpdate()) before allowing the user to update the other fields.

Reply:
Would this work?
```suggestion
      if (authUser.firebaseUser.email !== data.email) {
        await FirebaseService.doEmailUpdate(data.password, data.email);
      } else {
        await FirebaseService.doReauthenticateUser(data.password);
      }
```